languages/go/util.go

```diff
@@ -0,0 +1,32 @@
+package sdk
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+func checkSuccessAndError(responseStr string, v interface{}) error {
+	var wrapper struct {
+		Success      bool    `json:"success"`
+		ErrorMessage *string `json:"errorMessage"`
+	}
+
+	err := json.Unmarshal([]byte(responseStr), &wrapper)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal wrapper response: %v", err)
+	}
+
+	if !wrapper.Success {
+		if wrapper.ErrorMessage != nil {
+			return fmt.Errorf("API error: %s", *wrapper.ErrorMessage)
+		}
+		return fmt.Errorf("API error: unknown")
+	}
+
+	err = json.Unmarshal([]byte(responseStr), &v)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal response: %v", err)
+	}
+
+	return nil
+}
```

Comment:
I don't think this is correct, the responses we get from the API are of the format `{ success, errorMessage, data }`. We're handling the success and errorMessage correctly, but once that's done we try to decode the whole message into `&v`, while we should only be decoding the data field, something like this:
```go
func checkSuccessAndError(responseStr string, v interface{}) error {
	var wrapper struct {
		Success      bool             `json:"success"`
		ErrorMessage *string          `json:"errorMessage"`
		Data         *json.RawMessage `json:"data"`
	}

	err := json.Unmarshal([]byte(responseStr), &wrapper)
	if err != nil {
		return fmt.Errorf("failed to unmarshal wrapper response: %v", err)
	}

	if !wrapper.Success {
		if wrapper.ErrorMessage != nil {
			return fmt.Errorf("API error: %s", *wrapper.ErrorMessage)
		}
		return fmt.Errorf("API error: unknown")
	}

	err = json.Unmarshal(*wrapper.Data, &v)
	if err != nil {
		return fmt.Errorf("failed to unmarshal response: %v", err)
	}

	return nil
}
```

Reply:
This , looks correct. I made the changes